File: java/src/com/stockmarket/core/Logger.java
package com.stockmarket.core;

import java.io.FileWriter;
import java.io.IOException;
import java.text.DateFormat;
import java.text.SimpleDateFormat;
import java.util.Date;

public final class Logger {
    private static final String LOG_FILE_NAME = "log.txt";

    private Logger() {
    }

    public static void toLog(String txt, boolean toConsole) {
        DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
        Date date = new Date();
        String out = dateFormat.format(date) + " " + txt;
        if (toConsole) {
            System.out.println(txt);
            out += " (from console)";
        }
        try (FileWriter writer = new FileWriter(LOG_FILE_NAME, true)) {
            writer.write(out);
            writer.write(System.getProperty("line.separator"));
            writer.flush();
        } catch (IOException ex) {
            System.out.println(ex.getMessage());
        }
    }

    public static void toLog(String txt) {
        toLog(txt, true);
    }
}

File: kotlin/src/com/stockmarket/core/Logger.kt
package com.stockmarket.core

import java.io.FileWriter
import java.io.IOException
import java.text.DateFormat
import java.text.SimpleDateFormat
import java.util.Date

object Logger {
    private const val LOG_FILE_NAME = "log.txt"
    fun toLog(txt: String, toConsole: Boolean) {
        val dateFormat: DateFormat = SimpleDateFormat("yyyy/MM/dd HH:mm:ss")
        val date = Date()
        var out: String = dateFormat.format(date).toString() + " " + txt
        if (toConsole) {
            System.out.println(txt)
            out += " (from console)"
        }
        try {
            FileWriter(LOG_FILE_NAME, true).use({ writer ->
                writer.write(out)
                writer.write(System.getProperty("line.separator"))
                writer.flush()
            })
        } catch (ex: IOException) {
            System.out.println(ex.message)
        }
    }

    fun toLog(txt: String) {
        toLog(txt, true)
    }
}